test/distributed/test_device_mesh.py

```diff
@@ -834,6 +834,11 @@ def test_flatten_mesh_3d(self):
         # Test flatten contiguous dims
         dp_cp_mesh = mesh_3d["dp", "cp"]
         flattened_dp_cp_mesh = dp_cp_mesh._flatten()
+        with self.assertRaisesRegex(
+            RuntimeError,
+            "already exists for submesh of the DeviceMesh",
```

Comment:
i am confused. are you trying to test that we throw an error if someone flattens twice?  or are trying to enable flattening twice to work?  the PR desc suggested you want to enable flatten twice to work.